app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java

```diff
@@ -144,10 +141,39 @@ public void afterTextChanged(Editable s) {
         }
     }
 
+    /**
+     * Uses previously saved theme if there is any, if not then uses default.
+     * Adds preference theme listener and saves value chosen by user to shared preferences
+     * to remember later
+     */
+    private void prepareTheme() {
+
+        String currentTheme = getCurrentTheme();
+
+        themeListPreference.setSummary(getThemeSummary(currentTheme));
+        themeListPreference.setValue(currentTheme);
+
+        themeListPreference.setOnPreferenceChangeListener((preference, newValue) -> {
+            String userSelectedValue = (String) newValue;
+            themeListPreference.setSummary(getThemeSummary(userSelectedValue));
+            getActivity().recreate();
```

Comment:
the recreate of the activity should mean that the summary gets set by the code above us anyhow, or at least that is how the code is running in my head. try it with deleting `setSummary`

Reply:
You are right, I removed the unnecessary code!